full/src/test/java/apoc/redis/RedisTest.java

```diff
@@ -0,0 +1,375 @@
+                r -> assertEquals(List.of("prefix1", "foo", "bar"), fromCodecList((List<Object>) r.get("value"))));
+
+        TestUtil.testCall(db, "CALL apoc.redis.pop($uri, $listKey, $config)",
+                map("uri", URI, "listKey", listKey, "config", map("codec", this.codec, "right", false)),
+                r -> assertEquals("prefix1", fromCodec(r.get("value"))));
+
+        TestUtil.testCall(db, "CALL apoc.redis.lrange($uri, $myListKey, 0 , 10, $config)",
+                map("uri", URI, "myListKey", listKey, "config", config),
+                r -> assertEquals(List.of("foo", "bar"), fromCodecList((List<Object>) r.get("value"))));
+    }
+
+    @Test
+    public void testSetsCommands() {
+        Map<String, Object> config = map("codec", this.codec);
+        Object key = getByCodec("mySetKey");
+        TestUtil.testCall(db, "CALL apoc.redis.sadd($uri, $key, $members, $config)",
+                map("uri", URI, "key", key, 
+                        "members", getListByCodec(List.of("foo", "bar", "baz")), "config", config),
+                r -> assertEquals(3L, r.get("value")));
+
+        TestUtil.testCall(db, "CALL apoc.redis.sadd($uri, $mySetKeyTwo, $members, $config)",
+                map("uri", URI, "members", getListByCodec(List.of("alpha", "beta")),
+                        "mySetKeyTwo", getByCodec("mySetKeyTwo"), "config", config),
+                r -> assertEquals(2L, r.get("value")));
+
+        TestUtil.testCall(db, "CALL apoc.redis.sunion($uri, $keys, $config)",
+                map("uri", URI, "keys", getListByCodec(List.of("mySetKey", "mySetKeyTwo")), "config", config),
+                r -> assertEquals(Set.of("foo", "bar", "baz", "alpha", "beta"), new HashSet<>(fromCodecList((List<Object>) r.get("value")))));
+
+        TestUtil.testCall(db, "CALL apoc.redis.scard($uri, $key, $config)",
+                map("uri", URI, "key", key, "config", config),
+                r -> assertEquals(3L, r.get("value")));
+
+        TestUtil.testCall(db, "CALL apoc.redis.smembers($uri, $key, $config)",
+                map("uri", URI, "key", key, "config", config),
+                r -> assertEquals(Set.of("foo", "bar", "baz"), new HashSet<>(fromCodecList((List<Object>) r.get("value")))));
+
+        TestUtil.testCall(db, "CALL apoc.redis.spop($uri, $key, $config)",
+                map("uri", URI, "key", key, "config", config),
+                r -> assertTrue(List.of("foo", "bar", "baz").contains(fromCodec(r.get("value")))));
+
+        TestUtil.testCall(db, "CALL apoc.redis.scard($uri, $key, $config)",
+                map("uri", URI, "key", key, "config", config),
+                r -> assertEquals(2L, r.get("value")));
+    }
+
+    @Test
+    public void testSortedSetsCommands() {
+        Map<String, Object> config = map("codec", this.codec);
+        Object key = getByCodec("mySortedSetKey");
+        TestUtil.testCall(db, "CALL apoc.redis.zadd($uri, $key, [0, $first, 100, $third, 1, $second], $config)",
```

Comment:
Certainly, it would be more readable, but unfortunately I cannot transform the parameter into a map because I would have the error `type Map cannot be converted to a Neo4j type: Maps are required to have String keys`, with `byte[]` codec.

Reply:
Ok, makes sense. Thanks for clarifying!